CMakeLists.txt

```diff
@@ -38,6 +38,7 @@ endif()
 # Options and settings
 #-------------------------------------------------------------------------------
 option(STABLEHLO_ENABLE_BINDINGS_PYTHON "Enables StableHLO Python bindings" OFF)
+option(STABLEHLO_STRICT_LINT "Build StableHLO with strict warnings and warnings as errors" OFF)
```

Comment:
How about `STABLEHLO_ENABLE_STRICT_BUILD`?
  1) To be consistent with another `STABLEHLO_ENABLE_*` option above.
  2) I think `BUILD` is a better fit thant `LINT` because even the description of the option says "lint".